### PR TITLE
Support indexing by Expr (e.g. chn[:(x[1])])

### PR DESCRIPTION
**Pull request by xukai92**

### CODE REVIEW

**Comment by xukai92** [comment]
I removed two piece of codes in https://github.com/yebai/Turing.jl/pull/295/commits/e8e67477650ff79fde0d3154c6a1186b94be4725. I think they are not used any more. How do you think?

**Comment by xukai92** [comment]
@yebai Do you think https://github.com/yebai/Turing.jl/commit/e8e67477650ff79fde0d3154c6a1186b94be4725 is fine? If so let's merge this PR.

File: src/core/util.jl [hunk 1]
```diff
@@ -66,3 +66,8 @@ align(x,y) = begin
 
   (x,y)
 end
+
+macro sym_str(var)
```
Comment (yebai) [inline src/core/util.jl]: Perhaps we can rename this to ‘sym2str’?

File: src/core/io.jl [hunk 2]
```diff
@@ -175,6 +175,8 @@ function Base.getindex(c::Chain, v::Symbol)
   end
 end
 
+Base.getindex(c::Chain, expr::Expr) = c[Symbol(string(expr))]
```
Comment (yebai) [inline src/core/io.jl]: We need some sanity check here, e.g. verify ‘expr’ is an indexing expression in the form of ‘x[1]…’.

File: src/core/util.jl (hunk 1 above)
Comment (xukai92) [inline src/core/util.jl]: This is a trick I just learnt. If we write the macro name as `xxx_str` we can directly call it by `xxx"something"`. In our case, we can create a symbol by `sym"x[1]"`.

File: src/core/io.jl (hunk 2 above)
Comment (xukai92) [inline src/core/io.jl]: I see it makes sense!

File: src/core/util.jl (hunk 1 above)
Comment (yebai) [inline src/core/util.jl]: Oh, I see.

**Pull request merged**